demo/index.html

```diff
@@ -831,27 +831,35 @@
               compact: false,
               pageSizeOptions: [1, 10, 50],
               pageSize: 2,
-              feed: hx.dataTable.objectFeed({
-                headers: [
-                  { name: 'Name', id: 'name', groups: ['Child Group 1', 'Parent Group'] },
-                  { name: 'Age', id: 'age', groups: ['Child Group 1', 'Parent Group'] },
-                  { name: 'Profession', id: 'profession', groups: ['Child Group 2', 'Parent Group'] }
-                ],
-                rows: [
-                  {
-                    id: 0, // hidden details can go here (not in the cells object)
-                    cells: { 'name': 'Bob', 'age': 25, 'profession': 'Developer' }
-                  },
-                  {
-                    id: 1,
-                    cells: { 'name': 'Jan', 'age': 41, 'profession': 'Artist' }
-                  },
-                  {
-                    id: 2,
-                    cells: { 'name': 'Dan', 'age': 41, 'profession': 'Builder' }
-                  }
-                ]
-              })
+              feed: {
+                headers: function (cb) {
+                  cb([
+                    { name: 'Name', id: 'name', groups: ['Child Group 1', 'Parent Group'] },
+                    { name: 'Age', id: 'age', groups: ['Child Group 1', 'Parent Group'] },
+                    { name: 'Profession', id: 'profession', groups: ['Child Group 2', 'Parent Group'] }
+                  ])
+                },
+                rows: function (range, cb) {
+                  cb({
+                    rows: [
+                      {
+                        id: 0, // hidden details can go here (not in the cells object)
+                        cells: { 'name': 'Bob', 'age': 25, 'profession': 'Developer' }
+                      },
+                      {
+                        id: 1,
+                        cells: { 'name': 'Jan', 'age': 41, 'profession': 'Artist' }
+                      },
+                      {
+                        id: 2,
+                        cells: { 'name': 'Dan', 'age': 41, 'profession': 'Builder' }
+                      }
+                    ].slice(range.start, range.end + 1),
+                    filteredCount: range.end >= 3 ? 3 : undefined
```

Comment:
Just to clarify, the problem here is that when the page size cleanly divides the filtered count you can click the next button even when no data exists for the next page:

![data-table-pagination-example](https://cloud.githubusercontent.com/assets/11637484/18633400/14a1bec4-7e74-11e6-89e8-9a02586d8a47.gif)

The same happens if the page size is set to 3.

I think this is simply because range.end uses 0 based indexing, so `range.end >= 3` is actually checking if we've gone beyond the end of the 'stream' by 1, rather than checking if its at end. So for this example checking `range.end >=2` would resolve the problem. (If I've misunderstood how it works, please correct me :-) )

This would be the same in the iterator/stream situation, where you could check something like `iterator.hasNext()` before deciding to return a value for `filteredCount` to avoid the empty page problem.